CRM/Aoservicelisting/Form/ProviderApplicationForm.php

```diff
@@ -40,25 +42,30 @@ public function setDefaultValues() {
         $primaryContactPhone = civicrm_api3('Phone', 'getsingle', ['contact_id' => $this->_loggedInContactID, 'is_primary' => 1]);
         $regulatorUrlPresent = (!empty($primaryContact[REGULATED_URL]));
         $staffRowCount = $abaStaffCount = 1;
+        //If the primary contact has a regulated url variable set also set the first regulated staff details to match that of the primary contact.
         if ($regulatorUrlPresent) {
           $defaults['staff_first_name['. $staffRowCount . ']'] = $defaults['primary_first_name'] = $primaryContact['first_name'];
           $defaults['staff_last_name['. $staffRowCount . ']'] = $defaults['primary_last_name'] = $primaryContact['last_name'];
           $defaults['staff_contact_id['. $staffRowCount . ']'] = $this->_loggedInContactID;
           $defaults['staff_record_regulator[' . $staffRowCount . ']'] = $primaryContact[REGULATED_URL];
           $staffRowCount++;
         }
+        // if the primary contact has a ABA certificate field set then prefill the first ABA staff member details with teh primary contact details. 
         if (!empty($primaryContact[CERTIFICATE_NUMBER])) {
-          $defaults['aba_first_name[' . $abaStaffCount . ']'] = $primaryContact['first_name'];
-          $defaults['aba_last_name[' . $abaStaffCount . ']'] = $primaryContact['last_name'];
+          $defaults['aba_first_name[' . $abaStaffCount . ']'] = $defaults['primary_first_name'] = $primaryContact['first_name'];
```

Comment:
@Edzelopez oh look another bug, if the primary contact had been an ABA staff member the primary details wouldn't have filled in on edit i suspect

Reply:
yup, good catch